test/forms/mappings/ConstraintsSpec.scala

```diff
@@ -156,4 +156,44 @@ class ConstraintsSpec extends WordSpec with MustMatchers with Constraints with R
 
     behave like regexWithValidAndInvalidExamples(companyUniqueTaxReference, validUtr, invalidUtr, invalidMsg, utr)
   }
+
+  "email" must {
+
+    val validEmail = Table(
+      "a@email.com",
+      "a@bc",
+      "123@456"
+    )
+
+    val invalidEmail = Table(
+      "value",
+      "32423423432423",
+      "12323",
+      "@@@@@@"
+    )
+
+    val invalidMsg = "contactDetails.error.email.valid"
+
+    behave like regexWithValidAndInvalidExamples(emailAddress, validEmail, invalidEmail, invalidMsg, email)
+  }
+
+  "wholeNumber" must {
+
+    val validNumber = Table(
+      "1",
+      "99999999999999999999999",
+      "123456"
+    )
+
```

Comment:
Same as validEmail Table. Add a first row that is the name of the data item, eg "number". Make invalidNumber the same.